torch/csrc/jit/passes/quantization.cpp

```diff
@@ -716,6 +719,150 @@ c10::optional<IValue> toTwoElementIntList(Value* v) {
+    for (Value* v : uses_to_rewrite_) {
+      const auto& path = value_to_path_map_.at(v);
+      const auto& m = findChildModule(module_, path);
+      // add a clone of the child module to the parent of the duplicated module
+      const auto& child_name = addChildModule(module_, m, path);
+      TORCH_INTERNAL_ASSERT(v->node()->kind() == prim::GetAttr);
+      // change the name in GetAttr call
+      auto original_name = v->node()->s(attr::name);
+      v->node()->s_(attr::name, child_name);
+      GRAPH_UPDATE(
+          "Module use dedup: changing use of original module ",
+          original_name,
+          " to ",
+          child_name);
+    }
+  }
+
+  script::Module findChildModule(
+      const script::Module& module,
+      const std::vector<std::string>& path) {
+    script::Module m = module;
+    for (size_t i = 0; i < path.size(); ++i) {
+      m = m.attr(path[i]).toModule();
+    }
+    return m;
+  }
+
+  std::string addChildModule(
+      script::Module module,
+      const script::Module& child_module,
+      const std::vector<std::string>& path) {
+    TORCH_INTERNAL_ASSERT(
+        path.size() > 0, "path must have at least one element.");
+    // Parent module of the leaf child module corresponding to
+    // the path
+    auto parent_of_leaf = findChildModule(
+        module, std::vector<std::string>(path.begin(), path.end() - 1));
+
+    // Original name of the child module
+    std::string original_name = path[path.size() - 1];
+    int uid = 0;
+    std::string child_name = original_name + "_" + c10::to_string(uid++);
+    while (parent_of_leaf.hasattr(child_name)) {
+      child_name = original_name + "_" + c10::to_string(uid++);
+    }
+    parent_of_leaf.register_module(child_name, child_module.clone_instance());
+    return child_name;
+  }
+
+  // Check if value is the input of the graph
+  bool hitGraphInput(Value* value) {
+    Graph* graph = value->owningGraph();
+    const auto& inputs = graph->inputs();
+    return std::find(inputs.begin(), inputs.end(), value) != inputs.end();
+  }
+
+  Value* getModuleAccessPath(Value* instance, std::vector<std::string>& path) {
+    // Iterator to traverse back the GetAttr calls
+    Value* iter = instance;
+    // trace back the instance to recover the path of the submodule
+    while (!hitGraphInput(iter) && iter->node()->kind() == prim::GetAttr) {
+      Node* get_attr = iter->node();
+      // record the name of GetAttr
+      path.push_back(get_attr->s(attr::name));
+      // trace back the chain of GetAttr
+      iter = get_attr->inputs()[0];
+    }
+    return iter;
+  }
+
+  script::Module module_;
+  // Map from value of module instance to the list of names of submodules
+  // starting from the top level module, e.g. ["sub1", "sub2", "relu"]
+  // Also this is a cache of calling `getModuleAccessPath` of the value
+  std::unordered_map<Value*, std::vector<std::string>> value_to_path_map_;
+  std::unordered_set<script::ModulePtr> module_set_;
```

Comment:
This variable name just restates the type (i.e. set of module). It would be better to name it for what it is semantically used for.